### PR TITLE
Updated scripts for various tool installation.

### PR DESCRIPTION
**Pull request by jobyko**
Adding scripts to install various tools.

### CODE REVIEW

**Comment by jobyko** [comment]
@asa1997 Please review and share your comments.

**Comment by jobyko** [comment]
@asa1997 . Arun, please test/review and share your comments.

@sumodgeorge

**Comment by asa1997** [comment]
@panickervinod @sumodgeorge 
@jobyko it would be good if you could also add an if-statement to check whether these tools are already present, if so, skip the installation and echo - the tools is already installed

File: artifact.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	
+	wget -qO - $PUBKY |  apt-key add - 
+	add-apt-repository "deb [arch=amd64] $REPOURL $(lsb_release -cs) main"
+	apt-get update
+	
+	echo "Installing Jfrog artifactory"
+	apt install jfrog-artifactory-oss -y
+	
+	echo "Setting up services!!!"
+	for i in start enable status ; do
```
Comment (asa1997) [inline artifact.sh:33]: Try and use a single variable here instead of mentioning start enable status.

File: artifact.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		systemctl $i artifactory.service
```
Comment (asa1997) [inline artifact.sh]: jfrog

script halts at status ( status was active). Will have to give a keyboard interruption.

Add a grep condition to check whether the status was active.

File: artifact.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+PUBKY="https://api.bintray.com/orgs/jfrog/keys/gpg/public.key"
+REPOURL="https://jfrog.bintray.com/artifactory-debs"
+
+ echo "Installing Java for Jfrog Artifactory"
```
Comment (asa1997) [inline artifact.sh]: Using coloured echo commands could greatly help at debugging the flow of the script.

File: artifact.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	
+	echo "Installing Jfrog artifactory"
+	apt install jfrog-artifactory-oss -y
+	
+	echo "Setting up services!!!"
+	for i in start enable status ; do
+		systemctl $i artifactory.service
+	done 
+	if [ $? -eq 0 ]; then
+		echo "Install succesfull"
```
Comment (asa1997) [inline artifact.sh]: could not find these outputs as the script halts at above status.

File: artifact.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ apt-get update
+ apt-get install default-jdk -y
+ if [ $? -eq 0 ]; then
+	
+	wget -qO - $PUBKY |  apt-key add - 
+	add-apt-repository "deb [arch=amd64] $REPOURL $(lsb_release -cs) main"
+	apt-get update
+	
+	echo "Installing Jfrog artifactory"
+	apt install jfrog-artifactory-oss -y
```
Comment (asa1997) [inline artifact.sh:30]: downloads about 405MB. Check whether unzipping the zip file is faster.

File: beEF.sh [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+if ! grep 'apt-get install -y' $INSTALL_DIR/install >/dev/null 2>&1; then
+  sed -i 's/apt-get install/apt-get install -y/' $INSTALL_DIR/install
+fi
+#
+(cd $INSTALL_DIR && echo y | ./install) &&
+sed -i -E 's/passwd: .*$/passwd: "password"/' $INSTALL_DIR/config.yaml
+	if [ $? -eq 0 ] ; then	
+		echo "Pre-requisites Installation completed successfully..."	
+	else	
+		echo "issue with Pre-requisites, Please check!!!"
```
Comment (asa1997) [inline beEF.sh]: Got this output, while testing.

File: beEF.sh [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+ fi
+	systemctl disable $INSTALL_NAME || echo >/dev/null # ingore failure
+	rm -fr $INSTALL_DIR $SYSTEMD_DIR/$INSTALL_NAME.service
+	systemctl daemon-reload
+	exit 0
+fi
+#
+check_artifact 
+#
+## Pre-requisites
```
Comment (asa1997) [inline beEF.sh]: check whether you could modularize these operations.

File: beEF.sh [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+systemctl restart beef
+if [ $? -ne 0 ]; then	
+	echo "package reload/enable failed"
+	exit 1 
+else
+	# success message
+	cat <<EOF
+		The Browser Exploitation Framework $INSTALL_VERSION installed successfully. Run following command to test
+		systemctl start $INSTALL_NAME
+		http://localhost:3000/ui/panel
```
Comment (asa1997) [inline beEF.sh]: Didnt get the ui 
beef_ui

File: PyPi.sh [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+	apt-get remove --purge python-pip
+	if [ $? -ne 0 ]; then
+		apt-get remove --purge python3-pip
+			if [ $? -ne 0 ]; then
+				echo "Uninstall Failed"
+			fi 
+	fi 
+fi
+###MAIN
+install_app
```
Comment (asa1997) [inline PyPi.sh:54]: pypi_terminal

Got this output. Do add a condition which checks whether a package is already present or not.

File: Selenium.sh [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+	exit 1 
+else
+	# success message
+	cat <<EOF
+	Selenium Standalone Server $INSTALL_VERSION installed successfully. Run following command to test
+
+	systemctl start $INSTALL_NAME
+	http://localhost:4444
+EOF
+fi
```
Comment (asa1997) [inline Selenium.sh:115]: selenium_ui

This is the UI that i got. The terminal o/p is below.

selenium_terminal

File: Selenium.sh [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+#
+## Pre-requisites
+echo " Pre-requisites Installation initiated..."
+apt update &&
+	apt install -y openjdk-8-jdk  &&
+	JAVA8_HOME=$(update-alternatives --list java | grep java-8 | sed -e 's/\/bin\/java//')
+if [ $? -eq 0 ] ; then	
+	echo "Pre-requisites Installation completed successfully..."	
+else	
+	echo "issue with Pre-requisites, Please check!!!"
```
Comment (asa1997) [inline Selenium.sh:68]: - Got this output while testing. 
- Could use a different colour.

File: Selenium.sh [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+  update-alternatives --remove $INSTALL_NAME $INSTALL_DIR/$INSTALL_NAME.sh
+  systemctl disable $INSTALL_NAME
+  rm -fr $INSTALL_DIR $SYSTEMD_DIR/$INSTALL_NAME.desktop
+  exit 0
+fi
+#
+check_artifact 
+#
+## Pre-requisites
+echo " Pre-requisites Installation initiated..."
```
Comment (asa1997) [inline Selenium.sh:61]: Could use a different colour.

File: Selenium.sh [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+# Ensure artifact is present
+check_artifact(){
+
+	if [ ! -f $ARTIFACT_TARBALL ]; then
+	
+		sudo mkdir -p $INSTALL_ARTIFACTS_DIR
+		if [ ! -z "$2" ]; then 
+			INSTALL_VERSION="$2"
+		fi
+		echo "Initiating installation process for Selenium version : $INSTALL_VERSION"
```
Comment (asa1997) [inline Selenium.sh:21]: Could use a different colour.

File: vscode.sh [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+	sudo add-apt-repository "deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main"
+
+	echo "install VScode"
+	sudo apt update &&
+		sudo apt install code
+}
+
+###Main
+
+install_app
```
Comment (asa1997) [inline vscode.sh:39]: vscode_terminal

Comment (asa1997) [reply vscode.sh:39]: - Got the above o/p. 
- Add a condition to check whether the tools and packages exists.

File: zap_tool.sh [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+## Pre-requisites
+echo " Pre-requisites Installation initiated..."
+sudo apt update && 
+	sudo apt install -y openjdk-8-jdk &&
+	JAVA8_HOME=$(update-alternatives --list java | grep java-8 | sed -e 's/\/bin\/java//')
+	if [ $? -eq 0 ] ; then	
+		echo "Pre-requisites Installation completed successfully..."	
+	else	
+		echo "issue with Pre-requisites, Please check!!!"
```
Comment (asa1997) [inline zap_tool.sh:66]: Got this output.

**Review by asa1997 — COMMENTED** [review]
@panickervinod @sumodgeorge 
@jobyko A few comments,

- Do use a coloured echo commands, whenever necessary, to make it easier for debugging the running process.
- Do add the condition, wherever required, to check whether a package/tool is already present or not.

**Pull request merged**